### PR TITLE
[WASM] Improve debugging support

### PR DESCRIPTION
**Pull request by jeromelaban**
- Add support for mono PDBs and basic Source Link support
- Add support for local files
- Fix NRE when some method tokens cannot be found
- Add support for full document path (fixes duplicate file names debugging)
- Fix race condition in debugger messages
- Don't fail on multi-document methods

### CODE REVIEW

**Comment by vargaz** [comment]
Some comments:
- please use the mono coding conventions, i.e. K&R indentation, spaces, etc.:
https://www.mono-project.com/community/contributing/coding-guidelines/
- the indentation seems to be off in a lot of places, probably spaces vs tabs.

**Comment by jeromelaban** [comment]
Thanks, I updated the .editorconfig file to help align with the conventions.

**Comment by lewing** [comment]
@monojenkins build failed

File: sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs [hunk 1]
```diff
@@ -640,23 +646,42 @@ void OnCompileDotnetScript (int msg_id, CancellationToken token)
...
+					var doc = await new WebClient ().DownloadStringTaskAsync (src_file.OriginalSourcePath);
+					res.Write (doc);
+
+					var o = JObject.FromObject (new {
+						scriptSource = res.ToString ()
+					});
+
+					SendResponse (msg_id, Result.Ok (o), token);
+				}
+			} catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs]: Indeed, I added the message.

**Comment by jeromelaban** [comment]
Sure, I'll add a fallback if a local file is available.

**Comment by lewing** [comment]
Does it make sense to prefer the local file if available?  Feels like it is more likely to be the actual source if it exists?

**Comment by jeromelaban** [comment]
The scenario I'm referring to is when Source Link info is present (e,g, the assembly was built locally). A local file is available, but because there's a source link as well, and it's used first.

Maybe the change is to check for a local file first, then a Source Link file.

**Comment by jeromelaban** [comment]
There, it should be better, and I found an issue with the SourceLink resolution, it's now using all available mappings instead of just the first...

A local file is now preferred over the SourceLink one, if it exists.

**Comment by akoeplinger** [comment]
@monojenkins whitelist

**Comment by jeromelaban** [comment]
Nice, all checks have passed :)

**Pull request merged**